src/validationinterface.cpp

```diff
@@ -406,49 +442,74 @@ void ThreadNotifyWallets(CBlockIndex *pindexLastTip)
 
         // Notify block connections
         while (!blockStack.empty()) {
-            auto blockData = blockStack.back();
-            blockStack.pop_back();
-
-            // Read block from disk.
-            CBlock block;
-            if (!ReadBlockFromDisk(block, blockData.pindex, chainParams.GetConsensus())) {
-                LogPrintf(
-                        "*** %s: Failed to read block %s while notifying wallets of block connects",
-                        __func__, blockData.pindex->GetBlockHash().GetHex());
-                uiInterface.ThreadSafeMessageBox(
-                    strprintf(_("Error: A fatal internal error occurred, see %s for details"), GetDebugLogPath()),
-                    "", CClientUIInterface::MSG_ERROR);
-                StartShutdown();
-                return;
+            // Add more block connections to batchScanners until we have either
+            // added the rest of blockStack, or have reached the memory limit
+            // again. At this point, we know that blockStack.rbegin() is before
+            // or equal to blockStackScanned.
```

Comment:
"Before" is confusing here.

Reply:
Altered the comment to better convey what I meant.